spring-cloud-task-core/src/test/java/org/springframework/cloud/task/listener/TaskLifecycleListenerTests.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright 2015 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.task.listener;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+import ch.qos.logback.core.Appender;
+import org.junit.Test;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.PropertyPlaceholderAutoConfiguration;
+import org.springframework.boot.context.event.ApplicationFailedEvent;
+import org.springframework.cloud.task.util.TestDefaultConfiguration;
+import org.springframework.cloud.task.util.TestVerifierUtils;
+import org.springframework.context.annotation.AnnotationConfigApplicationContext;
+import org.springframework.context.event.ContextClosedEvent;
+
+/**
+ * Verifies that the TaskLifecycleListener Methods record the appropriate log header entries and
+ * result codes.
+ *
+ * @author Glenn Renfro
+ * @author Michael Minella
+ */
+public class TaskLifecycleListenerTests {
+
+	@Autowired
+	private TaskLifecycleListener listener;
+
+	@Test
+	public void testTaskCreate() {
+		AnnotationConfigApplicationContext createContext = new AnnotationConfigApplicationContext();
+		createContext.register(TestDefaultConfiguration.class, PropertyPlaceholderAutoConfiguration.class);
```

Comment:
We can create a `@Before` method to handle the initializtion of the context and the registration of the configuration classes (i.e. lines 48-49, 63-64, 78-79).  Then a `@After` method to close the context.